src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultDisplayTemplates.cs

```diff
@@ -25,12 +26,12 @@ public static string BooleanTemplate(IHtmlHelper htmlHelper)
 
             return htmlHelper.ViewData.ModelMetadata.IsNullableValueType ?
                 BooleanTemplateDropDownList(htmlHelper, value) :
-                BooleanTemplateCheckbox(value ?? false);
+                BooleanTemplateCheckbox(value ?? false, htmlHelper.HtmlEncoder);
         }
 
-        private static string BooleanTemplateCheckbox(bool value)
+        private static string BooleanTemplateCheckbox(bool value, IHtmlEncoder htmlEncoder)
```

Comment:
not a substantial refactor since the template `Func`s all have the `IHtmlHelper` parameter.  make sure the necessary encoder is a prop on `IHtmlHelper` and use that instead.  BTW it's useless to make anything `public` on `HtmlHelper` unless it's also in the interface.